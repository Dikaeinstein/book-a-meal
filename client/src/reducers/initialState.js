const initialState = {
  urls: [
    {
      id: 1,
      name: 'Signin',
      link: 'signin',
    },
    {
      id: 2,
      name: 'Signout',
      link: 'signout',
    },
  ],
  menus: {
    isFetching: false,
    data: {
      menu: {
        name: 'Menu for today',
        meals: [],
      },
      menus: [],
    },
    isSet: false,
    isSaving: false,
    fetchError: null,
    saveError: null,
    updateError: null,
    isUpdating: false,
  },
  user: {
    isSubmitting: false,
    loggedIn: false,
    data: {
      id: null,
      name: '',
      email: '',
      role: '',
    },
    error: null,
  },
  orders: {
    isFetching: false,
    isSaving: false,
    isUpdating: false,
    isDeleting: false,
    data: {
      userOrders: [],
      checkedOutOrder: {},
    },
    fetchError: null,
    saveError: null,
    updateError: null,
    deleteError: null,
  },
  meals: {
    isFetching: false,
    isUpdating: false,
    isSaving: false,
    data: [],
    fetchError: null,
    saveError: null,
    updateError: null,
  },
  dashboard: {
    isFetchingTotalOrders: false,
    isFetchingTotalSales: false,
    isUpdating: false,
    isDeleting: false,
    fetchTotalOrdersError: null,
    fetchTotalSalesError: null,
    isFetchingAllOrders: false,
    fetchAllOrdersError: null,
    deleteOrderError: null,
    updateOrderError: null,
    totalAmount: 0,
    totalOrders: 0,
    allOrders: [],
  },
};

export default initialState;
