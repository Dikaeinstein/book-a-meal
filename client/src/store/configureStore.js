import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
