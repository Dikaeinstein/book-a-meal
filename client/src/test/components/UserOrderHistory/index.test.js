import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedUserOrderHistory, { UserOrderHistory }
  from '../../../components/UserOrderHistory';
import initialState from '../../../reducers/initialState';


describe('<UserOrderHistory />', () => {
  it('should call handleRetry on when fetch orders fail', () => {
    const spy = jest.spyOn(UserOrderHistory.prototype, 'handleRetry');
    const wrapper = shallow(<UserOrderHistory
      fetchUserOrders={jest.fn()}
      isFetching={false}
      error={new Error()}
      currentUrl="http://testurl"
    />);
    wrapper.instance().handleRetry();
    expect(spy).toHaveBeenCalled();
  });
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const wrapper = shallow(<ConnectedUserOrderHistory
      store={store}
      fetchUserOrders={jest.fn()}
    />);
    expect(wrapper.props().isFetching).toBeFalsy();
    expect(wrapper.props().error).toBeNull();
  });
  it('should match snapshot', () => {
    const tree = shallow(<UserOrderHistory
      fetchUserOrders={jest.fn()}
      isFetching
      currentUrl="http://testurl"
    />);
    expect(tree).toMatchSnapshot();
  });
});
