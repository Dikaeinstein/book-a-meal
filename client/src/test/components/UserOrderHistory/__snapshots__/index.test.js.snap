// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserOrderHistory /> should match snapshot 1`] = `
<div
  className="loader-container"
>
  <Loading
    text="fetching orders..."
  >
    <Preloader
      active={true}
      color="blue"
      flashing={true}
      size="medium"
    />
  </Loading>
</div>
`;
